modules/kafka/src/main/java/org/testcontainers/containers/KafkaContainerCluster.java

```diff
@@ -0,0 +1,128 @@
+        }
+        if (internalTopicsRf < 0 || internalTopicsRf > brokersNum) {
+            throw new IllegalArgumentException("internalTopicsRf '" + internalTopicsRf + "' must be less than brokersNum and greater than 0");
+        }
+
+        this.brokersNum = brokersNum;
+        this.network = Network.newNetwork();
+
+        this.zookeeper = new GenericContainer("confluentinc/cp-zookeeper:" + confluentPlatformVersion)
+            .withNetwork(network)
+            .withNetworkAliases("zookeeper")
+            .withEnv("ZOOKEEPER_CLIENT_PORT", String.valueOf(KafkaContainer.ZOOKEEPER_PORT));
+
+        this.brokers = IntStream
+            .range(0, this.brokersNum)
+            .mapToObj(brokerNum ->
+                new KafkaContainer(confluentPlatformVersion)
+                    .withNetwork(this.network)
+                    .withNetworkAliases("broker-" + brokerNum)
+                    .dependsOn(this.zookeeper)
+                    .withExternalZookeeper("zookeeper:" + KafkaContainer.ZOOKEEPER_PORT)
+                    .withEnv("KAFKA_BROKER_ID", brokerNum + "")
+                    .withEnv("KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR", internalTopicsRf + "")
+                    .withEnv("KAFKA_OFFSETS_TOPIC_NUM_PARTITIONS", internalTopicsRf + "")
+                    .withEnv("KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR", internalTopicsRf + "")
+                    .withEnv("KAFKA_TRANSACTION_STATE_LOG_MIN_ISR", internalTopicsRf + "")
+            )
+            .collect(Collectors.toList());
+    }
+
+    public Network getNetwork() {
+        return this.network;
+    }
+
+    public GenericContainer getZooKeeper() {
+        return this.zookeeper;
+    }
+
+    public Collection<KafkaContainer> getBrokers() {
+        return this.brokers;
+    }
+
+    public String getBootstrapServers() {
+        return brokers.stream()
+            .map(KafkaContainer::getBootstrapServers)
+            .collect(Collectors.joining(","));
+    }
+
+    private Stream<GenericContainer> allContainers() {
+        Stream<GenericContainer> genericBrokers = this.brokers.stream().map(b -> (GenericContainer) b);
+        Stream<GenericContainer> zookeeper = Stream.of(this.zookeeper);
+        return Stream.concat(genericBrokers, zookeeper);
+    }
+
+    @Override
+    @SneakyThrows
+    public void start() {
+        Stream<Startable> startables = this.brokers.stream().map(b -> (Startable) b);
+        Startables.deepStart(startables).get(60, SECONDS);
+
+        Unreliables.retryUntilTrue(30, TimeUnit.SECONDS, () -> this.brokers.stream().findFirst()
+                .map(this::clusterBrokers)
+                .filter(brokers -> brokers.split(",").length == this.brokersNum)
+                .isPresent());
+    }
+
+    @SneakyThrows
+    private String clusterBrokers(KafkaContainer c) {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        dockerClient
```

Comment:
Have you considered using `zookeeper.execInContainer(cmd)`?

Also, what would be the use of this?

Reply:
It's used in `start()` to assert that Kafka cluster has formed, by checking ZooKeeper (its coordinator) to see if the right number of brokers have joined together.

Good idea about using the ZooKeeper container.  It looks like the `zookeeper-shell` script is present on the path there as well.  I've pushed an update.